### PR TITLE
cmd/flux-jobs: limit attributes requested

### LINKED ISSUE #2637: flux-jobs: limit attributes requested
As more attributes become available in `job-info`, the `jobs` list that is returned from `job-info.list` will increase in size b/c in `flux-jobs` we request EVERY attribute for every job.  This naturally can become heavy weight.

In the future, we'll want to limit this to only the attributes we care about for that particular listing.

May be something to do along side #2624 since we'll be parsing the format string anyways.

### PR DESCRIPTION
**Pull request by chu11**
Problem: flux-jobs requests all attribute information from the job-info list service, regardless of the information that is actually needed by the user.  As the job-info list service grows to support more
information, this can become excessive and lead to very large RPC responses.

Solution: Limit the attributes requested to only those attributes needed based on the output format input by the user.

Fixes #2637

Notes:

- was a little trickier than I had thought it would be.  Some of the format fields require multiple attributes to be returned from job-info, so I did a Python dict from field name to tuple of attributes (majority of which contain a single element).
- to my surprise Python did not appear to have a builtin `uniq()` like in Perl.  So I instead used sets to built up the "list" of attributes required without duplicates.

### CODE REVIEW

**Comment by grondo** [comment]
> to my surprise Python did not appear to have a builtin uniq() like in Perl. So I instead used sets to built up the "list" of attributes required without duplicates.

Nice, that came out pretty clean IMO!

**Comment by grondo** [comment]
I wonder if there is any way to test? Maybe just the fact that the existing tests pass is sufficient that nothing broke here, but at least one test that verifies that limited attributes are requested.. but I'm not sure that is viable. Eh, doesn't seem that important

**Comment by chu11** [comment]
> I wonder if there is any way to test? Maybe just the fact that the existing tests pass is sufficient that nothing broke here, but at least one test that verifies that limited attributes are requested.. but I'm not sure that is viable. Eh, doesn't seem that important

Yeah, I couldn't come up with anything obvious that could work.  Did some debug `print()`s just to make sure it was doing what I thought it would be doing, but that was it.

File: src/cmd/flux-jobs.py [hunk 1]
```diff
@@ -93,7 +93,8 @@ def __init__(self, info_resp):
         d.update(info_resp)
 
         #  Rename "state" to "state_id" until returned state is a string:
-        d["state_id"] = d.pop("state")
+        if d.has_key("state"):
```
Comment (grondo) [inline src/cmd/flux-jobs.py]: (oops forgot to hit add comment)

Looks like `has_key` method is no longer available in Python 3. This [guide](https://portingguide.readthedocs.io/en/latest/dicts.html) suggests using `in` instead.

**Comment by chu11** [comment]
hmmm ... a bunch of builders failed b/c

```
flux-jobs: ERROR: 'dict' object has no attribute 'has_key'
flux-jobs: ERROR: 'dict' object has no attribute 'has_key'
flux-jobs: ERROR: 'dict' object has no attribute 'has_key'
flux-jobs: ERROR: 'dict' object has no attribute 'has_key'
```

I guess `has_key()` isn't portable between python versions.  Limited Googling says `in` is the portable way.  Re-pushed with that tweak.

File: src/cmd/flux-jobs.py (hunk 1 above)
Comment (chu11) [reply src/cmd/flux-jobs.py]: oh lol, you posted just before I did :P

**Pull request merged**